### PR TITLE
Added optional RC channel updater

### PR DESCRIPTION
**Pull request by c4ffein**

### CODE REVIEW

**Comment by c4ffein** [comment]
Implements #1141

File: parsec/core/gui/new_version.py [hunk 1]
```diff
@@ -47,28 +55,32 @@ def _fetch_json_releases():
...
-            return (latest_from_head, url)
+            return latest_from_head, url
 
-        latest_version = (0, 0, 0)
+        latest_version = Version("0.0.0")
         latest_url = ""
 
-        for release in json_releases:
-            try:
+        try:
```
Comment (touilleMan) [inline parsec/core/gui/new_version.py:63]: Why the switch between try and for ? given the except does a return I don't see the difference...

File: tests/core/gui/test_new_version.py [hunk 2]
```diff
@@ -399,4 +408,64 @@ def test_windows_update_newest_announced_is_draft():
...
+    assert smallcheck() == ((1, 9, 0), gui_check_version_url)
+
+
+@patch("parsec.core.gui.new_version.__version__", new="1.7.0")
+@patch(
+    "parsec.core.gui.new_version.urlopen",
+    new=urlopener(
+        head_version="1.9.0",
+        api_json={
+            "v1.9.0-rc3": {"draft": False, "prerelease": False, "32": True, "64": True},
```
Comment (touilleMan) [inline tests/core/gui/test_new_version.py:420]: is it expected to have `prerelease: False` for a rc release ? (and also `32: True, 64: True` ?)

File: parsec/core/cli/run.py [hunk 3]
```diff
@@ -22,12 +22,15 @@
     # Let the GUI handle the parsing of the url to display dialog on error
     @click.argument("url", required=False)
     @click.option("--diagnose", "-d", is_flag=True)
+    @click.option("--rc-update", "-r", is_flag=True)
```
[the remark this answers is not in the document]
Comment (c4ffein) [reply parsec/core/cli/run.py]: Removed

File: parsec/core/config.py [hunk 4]
```diff
@@ -130,6 +132,7 @@ def config_factory(
         gui_first_launch=gui_first_launch,
         gui_last_version=gui_last_version,
         gui_check_version_at_startup=gui_check_version_at_startup,
+        gui_rc_update_enabled=gui_rc_update_enabled,
```
[the remark this answers is not in the document]
Comment (c4ffein) [reply parsec/core/config.py]: Right, changed that

File: parsec/core/gui/new_version.py [hunk 5]
```diff
@@ -17,25 +17,33 @@
 from parsec.core.gui.ui.new_version_info import Ui_NewVersionInfo
 from parsec.core.gui.ui.new_version_available import Ui_NewVersionAvailable
 
+from misc.releaser import Version, RELEASE_REGEX
```
[the remark this answers is not in the document]
Comment (c4ffein) [reply parsec/core/gui/new_version.py:20]: Went for the cool kid, works well!

File: parsec/core/gui/new_version.py (hunk 1 above)
Comment (c4ffein) [reply parsec/core/gui/new_version.py:63]: As it was finally decided to make a return at the first json inconsistency, as it means that data can't be trusted and we should redirect the user to the latest version in any case, we might as well put the iteration on `json_releases` in the try block

Comment (c4ffein) [reply parsec/core/gui/new_version.py:63]: Also, it covers the highly unlikely case where `json_releases` is not iterable

File: tests/core/gui/test_new_version.py (hunk 2 above)
Comment (c4ffein) [reply tests/core/gui/test_new_version.py:421]: No, it is expected to have `prerelease: True` for a prerelease, from RC to alpha. But as doing a manual release as obligated by hardware signing is a boring, simple process, it is also expected that we might end up forgetting to set that tiny checkbox at the end of it one day or another. So this PR is made to work whether we forget that checkbox or not, and we also test that :)

File: parsec/core/gui/new_version.py (hunk 1 above)
Comment (touilleMan) [reply parsec/core/gui/new_version.py:67]: 👍 

(note we should also add a `except  JSONDecodeError` in `_fetch_json_releases` to protect against json inconsistency)

File: tests/core/gui/test_new_version.py (hunk 2 above)
Comment (touilleMan) [reply tests/core/gui/test_new_version.py:421]: Ha ! I didn't read the test's name 😄

File: parsec/core/gui/new_version.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (touilleMan) [reply parsec/core/gui/new_version.py]: I think we should also add packaging in the `requirements.txt`

File: parsec/core/gui/new_version.py (hunk 1 above)
Comment (c4ffein) [reply parsec/core/gui/new_version.py:67]: Also done

File: parsec/core/gui/new_version.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (c4ffein) [reply parsec/core/gui/new_version.py]: Also done

**Pull request merged**